### ISSUE TITLE
False-positive I0021: Useless suppression of 'line-too-long' (useless-suppression)

### ISSUE THREAD

**Issue by bersbersbers** [PyCQA/pylint#4212]
### Steps to reproduce

Given a file `a.py`:
```python
"""False-positive I0021: Useless suppression of 'line-too-long' (useless-suppression)"""
VAR = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"  # pylint:disable=line-too-long

```

### Current behavior
Result of `pylint --enable=useless-suppression a.py`:
```
************* Module a
a.py:2:0: I0021: Useless suppression of 'line-too-long' (useless-suppression)

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior
No output.

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.7.2
astroid 2.5.1
Python 3.8.8 (default, Feb 22 2021, 14:04:38) 
[GCC 7.5.0]
```

**Comment by Pierre-Sassoulas** [comment]
I can reproduce this, I think this is possibly a false negative on ``line-too-long``.

**Comment by Pierre-Sassoulas** [comment]
I can reproduce this, thank you.

**Comment by bersbersbers** [comment]
Here's another example:
```python
"""False-positive ungrouped-imports or useless-suppression."""
from re import sub

if 2 > 1:
    from pathlib import Path

    if 3 > 1:
        from re import match  # pylint:disable=ungrouped-imports


print(Path(), sub("", "", ""), match("", ""))
```

Assuming that `ungrouped-imports`, there is a false-positive `useless-suppression`. (Otherwise, there is a false-positive `ungrouped-imports`.)

**Comment by Pierre-Sassoulas** [comment]
Hi @bersbersbers in your example the import are grouped each in their own scope (isort would not change the file) so there should not be an ungrouped-imports and the useles suppression is not a false positive.

**Comment by bersbersbers** [comment]
Agreed. I found #2366 which _is_ another false-positive of `useless-suppression`, but I also filed #4360 for the false-positive `ungrouped-imports`.

**Comment by Pierre-Sassoulas** [comment]
Another example I just encountered in astroid:
```
************* Module astroid.node_classes
astroid/node_classes.py:864:0: I0021: Useless suppression of 'unused-variable' (useless-suppression)
```

```python
        # pylint: disable=unused-variable; doesn't understand singledispatch
        @_repr_tree.register(NodeNG)
        def _repr_node(node, result, done, cur_indent="", depth=1):
            """Outputs a strings representation of an astroid node."""
````

If you remove the disable you then get ``astroid/node_classes.py:864:8: W0612: Unused variable '_repr_node' (unused-variable)
``

**Comment by DanielNoord** [comment]
> Another example I just encountered in astroid:
> 
> ```
> ************* Module astroid.node_classes
> astroid/node_classes.py:864:0: I0021: Useless suppression of 'unused-variable' (useless-suppression)
> ```
> 
>  ```python
>         # pylint: disable=unused-variable; doesn't understand singledispatch
>         @_repr_tree.register(NodeNG)
>         def _repr_node(node, result, done, cur_indent="", depth=1):
>             """Outputs a strings representation of an astroid node."""
> ```
> 
> If you remove the disable you then get `astroid/node_classes.py:864:8: W0612: Unused variable '_repr_node' (unused-variable)`

This is a different issue and a different issue should be opened to track it. If you can give me a minimal working example to include in the tests I can work on fixing the false positive!
